src/downshift.js

```diff
@@ -268,6 +270,14 @@ class Downshift extends Component {
   internalSetState(stateToSet, cb) {
     let onChangeArg
     const onStateChangeArg = {}
+
+    if (stateToSet.hasOwnProperty('inputValue')) {
```

Comment:
Better be careful here. `stateToSet` can be a function. Also, `this.getStateAndHelpers()` will get us the _old_ state, not any of the new state. So let's move this down to right before `return nextState` and then we can do:

```javascript
this.props.onInputValueChange(
  stateToSet.inputValue,
  {...this.getStateAndHelpers(), ...nextState},
)
```

Thoughts?

Reply:
Wouldn't this cause `inputValue` being read from stale props again?

Reply:
Nope, because `nextState` will have the `inputValue`.

But actually, it should probably be this instead:

```javascript
this.props.onInputValueChange(
  stateToSet.inputValue,
  {...this.getStateAndHelpers(), ...stateToSet},
)
```

And we could put it right below `stateToSet.type = ...`

Reply:
[Tried that here](https://github.com/paypal/downshift/pull/230/commits/c7d998fc81835898437bae97d5b6ab308ba2352f), but placing the `onInputValueChange` inside the setState function, causes the children to re-render with the old value once, still leading to the cursor jumping to the end.